flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/rules/FlinkBatchRuleSets.scala

```diff
@@ -243,6 +245,7 @@ object FlinkBatchRuleSets {
     // scan optimization
     PushProjectIntoTableSourceScanRule.INSTANCE,
     PushFilterIntoTableSourceScanRule.INSTANCE,
+    PushPartitionIntoTableSourceScanRule.INSTANCE,
```

Comment:
in hep program, we could make sure the partition predicate could be push down totally, because its rules is part of `logical` volcano rules. this require the `PartitionableTableSource#getPartitions` returns remaining partition(first time, returns all partition). to make things easier, I will remove `PushPartitionIntoTableSourceScanRule` from volcano rule sets.